Riot/Modules/Home/AllChats/AllChatsViewController.swift

```diff
@@ -87,41 +104,230 @@ class AllChatsViewController: HomeViewController {
+    
+    private func showCreateSpace(parentSpaceId: String?) {
+        let coordinator = SpaceCreationCoordinator(parameters: SpaceCreationCoordinatorParameters(session: self.mainSession, parentSpaceId: parentSpaceId))
+        let presentable = coordinator.toPresentable()
+        self.present(presentable, animated: true, completion: nil)
+        coordinator.callback = { [weak self] result in
+            guard let self = self else {
+                return
+            }
+            
+            self.createSpaceCoordinator?.toPresentable().dismiss(animated: true) {
+                self.createSpaceCoordinator = nil
+                switch result {
+                case .cancel:
+                    break
+                case .done(let spaceId):
+                    self.switchSpace(withId: spaceId)
+                }
+            }
+        }
+        coordinator.start()
+        
+        self.createSpaceCoordinator = coordinator
+    }
+    
+    private func switchSpace(withId spaceId: String?) {
+        searchController.isActive = false
+
+        guard let spaceId = spaceId else {
+            self.dataSource.currentSpace = nil
+            updateUI()
+
+            return
+        }
+
+        guard let space = self.mainSession.spaceService.getSpace(withId: spaceId) else {
+            MXLog.warning("[AllChatsViewController] switchSpace: no space found with id \(spaceId)")
+            return
+        }
+        
+        self.dataSource.currentSpace = space
+        updateUI()
+        
+        self.recentsTableView.setContentOffset(.zero, animated: true)
+    }
+    
+    private func add(childCoordinator: Coordinator) {
+        self.childCoordinators.append(childCoordinator)
+    }
+    
+    private func remove(childCoordinator: Coordinator) {
+        self.childCoordinators.append(childCoordinator)
+    }
+    
+    private func showSpaceInvite() {
+        guard let session = mainSession, let spaceRoom = dataSource.currentSpace?.room else {
+            return
+        }
+        
+        let coordinator = ContactsPickerCoordinator(session: session, room: spaceRoom, initialSearchText: nil, actualParticipants: nil, invitedParticipants: nil, userParticipant: nil)
+        coordinator.delegate = self
+        coordinator.start()
+        add(childCoordinator: coordinator)
+        present(coordinator.toPresentable(), animated: true)
+    }
+    
+    private func showSpaceMembers() {
+        guard let session = mainSession, let spaceId = dataSource.currentSpace?.spaceId else {
+            return
+        }
+        
+        let coordinator = SpaceMembersCoordinator(parameters: SpaceMembersCoordinatorParameters(userSessionsService: UserSessionsService.shared, session: session, spaceId: spaceId))
+        coordinator.delegate = self
+        let presentable = coordinator.toPresentable()
+        presentable.presentationController?.delegate = self
+        coordinator.start()
+        add(childCoordinator: coordinator)
```

Comment:
Am I right in thinking this coordinator is removed by the presentation delegate method?

Reply:
Indeed. The coordinator is removed by the presentation delegate method or a the coordinator delegate method.